### PR TITLE
fix enum as struct field; fix errors in strict dynamic error

### PR DESCRIPTION
**Pull request by alanlzhang**
added `impl Default` for enum in rust

and `as int` for enum in dart

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Good job! I will review after CI goes green :)

**Comment by alanlzhang** [comment]
Finally CI goes green...

File: frb_codegen/src/main.rs [hunk 1]
```diff
@@ -49,6 +49,11 @@ mod tests {
         static ref LOGGER: () = init_logger(".", true).unwrap();
     }
 
+    // #[cfg(windows)]
+    // const DART: &str = "dart.bat";
+    // #[cfg(not(windows))]
+    const DART: &str = "dart";
```
Comment (fzyzcjy) [inline frb_codegen/src/main.rs:55]: nit: hmm so it is the same as the old code?

**Review by fzyzcjy — COMMENTED** [review]
(just briefly gone through it, will continue reviewing probably within a day)

**Review by fzyzcjy — APPROVED** [review]
Good fix! (will merge after all conversations are resolved)

File: frb_codegen/src/main.rs (hunk 1 above)
Comment (alanlzhang) [reply frb_codegen/src/main.rs:55]: On my machine, I have to use `dart.bat` to run tests successfully. 

But on CI windows machine, `dart.bat` causes `not found` error.

So I have to comment out `dart.bat` to pass the CI.

Maybe flutter version matters.

Comment (fzyzcjy) [reply frb_codegen/src/main.rs:55]: Hmm which flutter are you using?

Comment (alanlzhang) [reply frb_codegen/src/main.rs:55]: `flutter doctor` :

image

**Pull request merged**